<a href="https://colab.research.google.com/github/Ankittiwari21/APS_LAB/blob/main/APS_LAB_PROBABILITY_QUESTION_26_08_26.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import random
head=0
tail=0
number=int(input("Enter the number of times you want to perform the experiment : "))
for  i in range (number):
  toss=random.choice(['HEAD','TAIL'])
  if toss=="HEAD":
    head+=1
  else :
    tail+=1


# counting the probability
prob_of_head=head/number
prob_of_tail=tail/number
print("number of times head :",head)
print("number of times tail :",tail)
print("Probability of getting head : ",prob_of_head)
print("Probability of getting tail : ",prob_of_tail)




Enter the number of times you want to perform the experiment : 10000
number of times head : 4885
number of times tail : 5115
Probability of getting head :  0.4885
Probability of getting tail :  0.5115


here i am counting the total number of heads and tails using rendom.choice function then calculating the probability of head tail .                                                             
1.n=10

---


number of times head : 6

---


number of times tail : 4

---


Probability of getting head :  0.6

---


Probability of getting tail :  0.4

---

---



---



n=100

---


number of times head : 46

---


number of times tail : 54

---


Probability of getting head :  0.46

---


Probability of getting tail :  0.54

---

---



---
n=1000


---
number of times head : 479

---


number of times tail : 521

---


Probability of getting head :  0.479

---


Probability of getting tail :  0.521

---

---



---
n=10000


---
number of times head : 4885

---


number of times tail : 5115

---


Probability of getting head :  0.4885

---


Probability of getting tail :  0.5115

---

---


















In [ ]:
#question 2 = rolling a die 6k times
one,two,three,four,five,six=0,0,0,0,0,0
number=int(input("Enter the number of times you want to perform the experiment : "))
for i in range(number):
  roll=random.randint(1,6)
  if roll==1:
    one+=1
  elif roll==2:
    two+=1
  elif roll==3:
    three+=1
  elif roll==4:
    four+=1
  elif roll==5:
    five+=1
  elif roll==6:
    six+=1


# counting the probability
prob_of_one=one/number
prob_of_two=two/number
prob_of_three=three/number
prob_of_four=four/number
prob_of_five=five/number
prob_of_six=six/number
print("probabability of getting one :",prob_of_one)
print("probabability of getting two :",prob_of_two)
print("probabability of getting three :",prob_of_three)
print("probabability of getting four :",prob_of_four)
print("probabability of getting five :",prob_of_five)
print("probabability of getting six :",prob_of_six)



Enter the number of times you want to perform the experiment : 6000
probabability of getting one : 0.17183333333333334
probabability of getting two : 0.16766666666666666
probabability of getting three : 0.17133333333333334
probabability of getting four : 0.16033333333333333
probabability of getting five : 0.162
probabability of getting six : 0.16683333333333333


here using the random.ranint i am genrating the numbers between 1 to 6 both including and based on what i get i am assighning to the respective numbers to count .then i am calculating the probability using the formula .

---



In [ ]:
#calculation the union of the probability ad deciodning the wheteher the event are dependent or independent
one,six=0,0
number=int(input("Enter the number of times you want to perform the experiment : "))
for i in range(number):
  roll=random.randint(1,6)
  if roll==1:
    one+=1

  elif roll==6:
    six+=1


# counting the probability
prob_of_one=one/number

prob_of_six=six/number
print("probabability of getting one :",prob_of_one)

print("probabability of getting six :",prob_of_six)
print("the combine probabbilty of two events : ",prob_of_one+prob_of_six)

Enter the number of times you want to perform the experiment : 10000
probabability of getting one : 0.1705
probabability of getting six : 0.1688
the combine probabbilty of two events :  0.33930000000000005


Enter the number of times you want to perform the experiment : 10000

---


probabability of getting one : 0.1705

---


probabability of getting six : 0.1688

---


the combine probabbilty of two events :  0.33930000000000005

---

---





the event are disjoint beacase the probabbilty of getting one and six cannot appear at the same time on the single dice

In [ ]:
''' Addition rule for overlapping events
For one die roll, define: A=getting an even number A={2,4,6} and B=getting a
number greater than 3 B={4,5,6}
Verify: P(A∪B)=P(A)+P(B)−P(A∩B)'''
one,two,four,five,six=0,0,0,0,0
number=int(input("Enter the number of times you want to perform the experiment : "))
for i in range(number):
  roll=random.randint(1,6)
  if roll==1:
    one+=1
  elif roll==2:
    two+=1
  elif roll==4:
    four+=1
  elif roll==5:
    five+=1
  elif roll==6:
    six+=1


# counting the probability
even=two+four+six
greater_than_three=four+five+six
prob_of_four_six=(four+six)/number

prob_of_even=even/number
prob_of_greater_than_three=greater_than_three/number

print("probabability of getting even :",prob_of_even)

print("probabability of getting greater than 3 :",prob_of_greater_than_three)
print("probabability of getting four and six :",prob_of_four_six)
#verify the formula
print(f"probability of combine event. event A {prob_of_even} and event B {prob_of_greater_than_three} which is union of event A and event B is {prob_of_even+prob_of_greater_than_three-prob_of_four_six} ")


Enter the number of times you want to perform the experiment : 10
probabability of getting even : 0.5
probabability of getting greater than 3 : 0.6
probabability of getting four and six : 0.3
probability of combine event. event A 0.5 and event B 0.6 which is union of event A and event B is 0.8 


In [ ]:
'''
6. Multiplication rule for independent events
A coin is tossed and a die is rolled 10,000 times.
Define:
 obtaining Head
 obtaining an even number
Verify: P(A∩B)=P(A)P(B)
'''
two,four,six=0,0,0
head=0
tail=0
number=int(input("Enter the number of times you want to perform the experiment : "))
for i in range(number):
  roll=random.randint(1,6)
  toss=random.choice(['HEAD','TAIL'])
  if roll==2:
    two+=1
  elif roll==4:
    four+=1
  elif roll==6:
    six+=1
  elif toss=="HEAD":
    head+=1
  elif toss=="tail":
    tail+=1

event_A=two+four+six
event_B=head
prob_of_event_A=event_A/number
prob_of_event_B=event_B/number
print(f"the  intersection of the event A {prob_of_event_A} and Event B {prob_of_event_B} is  {prob_of_event_A*prob_of_event_B}  ")


Enter the number of times you want to perform the experiment : 6000
the  intersection of the event A 0.5026666666666667 and Event B 0.24683333333333332 is  0.12407488888888889  
